lib/cli/cliCompact.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+const CLICommand = require('./cliCommand.js');
+const CLIObjects = require('./cliObjects.js');
+const messages = require('./messages.js');
+const tools = require('../tools.js');
+const fs = require('fs');
+
+/** Command ioBroker compact ... */
+module.exports = class CLICompact extends CLICommand {
+
+    /** @param {import('./cliCommand.js').CLICommandOptions} options */
+    constructor(options) {
+        super(options);
+        this.config = JSON.parse(fs.readFileSync(tools.getConfigFileName(), 'utf8'));
+    }
+
+    /**
+     * Executes a command
+     * @param {any[]} args
+     */
+    execute(args) {
+        const { callback, showHelp } = this.options;
+        const command = args[0];
+        switch (command) {
+            case 'status':
+                return this.statusCompactModeHost();
+            case 'enable':
+            case 'on':
+                return this.setCompactModeHost(true);
+            case 'disable':
+            case 'off':
+                return this.setCompactModeHost(false);
+            default:
+                if (command && command.includes('.')) {
+                    if (args[1]) {
+                        switch (args[1]) {
+                            case 'status':
+                                return this.statusCompactModeInstance(command);
+                            case 'enable':
+                            case 'on':
+                                return this.setCompactModeInstance(command, true, args[2]);
+                            case 'disable':
+                            case 'off':
+                                return this.setCompactModeInstance(command, false);
+                            case 'group':
+                                return this.setCompactModeInstance(command, undefined, args[2]);
+                        }
+                    }
+                }
+
+                messages.error.unknownCommand('compact', command);
+                showHelp();
+                return void callback(3);
+        }
+    }
+
+    /**
+     * Output status of compact mode for host
+     */
+    statusCompactModeHost() {
+        const { callback } = this.options;
+
+        const compactMode = !!(this.config && this.config.system && this.config.system.compact);
+        console.log('Compact mode for this host is currently ' + (compactMode ? 'enabled' : 'disabled'));
+
+        return void callback();
+    }
+
+    /**
+     * Set compact mode for this host
+     * @param targetState target state to set for host
+     */
+    setCompactModeHost(targetState) {
+        const { callback, dbConnect } = this.options;
+
+        dbConnect((_objects, _states, isOffline) => {
+            const compactMode = !!(this.config && this.config.system && this.config.system.compact);
```

Comment:
I think `currentState` would be more logical. I had to think about what line 80 does.

Reply:
Done